src/mon/PGMonitor.cc

```diff
@@ -535,6 +578,10 @@ bool PGMonitor::preprocess_query(MonOpRequestRef op)
 
 bool PGMonitor::prepare_update(MonOpRequestRef op)
 {
+  if (mon->osdmon()->osdmap.require_osd_release >= CEPH_RELEASE_LUMINOUS) {
+    return false;
```

Comment:
There is no 'loop' per se, just a couple call sites in Monitor.cc.  We can move these calls to the caller but I don't think it makes things any better or worse.  I don't know that there is much more we can do to remove/disable this code until we don't have to support the upgrade case any more (or that we'd improve runtime performance by doing so).